pkg/network/opencontrail/controller_test.go

```diff
@@ -2112,3 +2112,151 @@ func TestServiceBeforeNamespace(t *testing.T) {
 	_, err := types.VirtualNetworkByName(client, "default-domain:newns:service-foo")
 	assert.NoError(t, err)
 }
+
+
+func TestDomainVariable(t *testing.T) {
+	kube := mocks.NewKubeClient()
+
+	client := new(contrail_mocks.ApiClient)
+	client.Init()
+
+	client.AddInterceptor("virtual-machine-interface", &VmiInterceptor{})
+	client.AddInterceptor("virtual-network", &NetworkInterceptor{})
+
+	allocator := new(mocks.AddressAllocator)
+
+	controller := NewTestController(kube, client, allocator, nil)
+
+	controller.config.DefaultDomain = "test-domain"
+
+	pod1 := &api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			Name:      "test-xz1",
+			Namespace: "testns",
+			UID:       kubetypes.UID(uuid.New()),
+			Labels: map[string]string{
+				"name": "testpod",
+			},
+		},
+	}
+	pod2 := &api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			Name:      "test-xz2",
+			Namespace: "testns",
+			UID:       kubetypes.UID(uuid.New()),
+			Labels: map[string]string{
+				"name": "client",
+				"uses": "x1",
+			},
+		},
+	}
+
+	service := &api.Service{
+		ObjectMeta: api.ObjectMeta{
+			Name:      "s1",
+			Namespace: "testns",
+			Labels: map[string]string{
+				"name": "x1",
+			},
+		},
+		Spec: api.ServiceSpec{
+			Selector: map[string]string{
+				"name": "testpod",
+			},
+			ClusterIP: "10.254.42.42",
+		},
+	}
+
+	glog.Infof("Domain name is %s", controller.config.DefaultDomain)
+	netnsProject := new(types.Project)
+	netnsProject.SetFQName("", []string{controller.config.DefaultDomain, "testns"})
+	client.Create(netnsProject)
+
+	allocator.On("LocateIpAddress", string(pod1.ObjectMeta.UID)).Return("10.0.0.1", nil)
```

Comment:
You don't need to mock the allocator calls if you let NewTestController create a real allocator.